src/reducers/breakpoints.js

```diff
@@ -250,6 +262,34 @@ function setCondition(state, action) {
   }
 }
 
+// Syncing Methods
+function _optimisticlyAddBreakpoint(state, breakpoint) {
+  const id = makeLocationId(breakpoint.location);
+  const updateOpts = {
+    loading: true
+  };
+
+  return state.setIn(["breakpoints", id], updateObj(breakpoint, updateOpts));
+}
+
+function _commitBreakpoint(state, breakpoint, overrides = {}) {
+  const location = overrides.location || breakpoint.location;
+  const id = makeLocationId(location);
+  const updatedOpts = { ...overrides, loading: false };
+  const updatedBreakpoint = updateObj(breakpoint, updatedOpts);
```

Comment:
I think `updateObj` has outlived it's purpose.
I'd also try to be more aggressive about what we want to do.

```js
function updateBreakpoint(state, newBreakpoint, oldBreakpoint) {
  const id = makeLocationId(newBreakpoint.location)
  state.setIn(["breakpoints", id], {
    ...breakpoint,
    ...overrides,
    loading: false
  });
}
```